service/history/ndc/workflow_resetter.go

```diff
@@ -779,6 +811,38 @@ func (r *workflowResetterImpl) reapplyEventsFromBranch(
 	return nextRunID, nil
 }
 
+// populateChildInitEventsAfterReset populates the childInitEventsAfterReset map with the init events of child workflows that are not yet completed.
+func (r *workflowResetterImpl) populateChildInitEventsAfterReset(childInitEventsAfterReset map[int64]*historypb.HistoryEvent, event *historypb.HistoryEvent) {
+	// nolint:exhaustive
+	switch event.GetEventType() {
+	// track the child if it has been initiated after reset
+	case enumspb.EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED:
+		childInitEventsAfterReset[event.GetEventId()] = event
+
+	// remove the child if it has reached a terminal state
+	case enumspb.EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_FAILED:
+		attr := event.GetStartChildWorkflowExecutionFailedEventAttributes()
+		delete(childInitEventsAfterReset, attr.InitiatedEventId)
```

Comment:
hmm InitiatedEventId could duplicate across runs.

Reply:
Good catch.
Now that I won't be tracking completed children, I'll directly construct the map using child ID strings. No need to track init events.

Reply:
This should not be an issue now.